src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -551,23 +568,19 @@ def load_status_file_components(self, initial_load=False):
 
         self.composite_logger.log_debug("Loading status file components [InitialLoad={0}].".format(str(initial_load)))
 
-        # Verify the status file exists - if not, reset status file
-        if not os.path.exists(self.status_file_path) and initial_load:
+        # Verify the complete status file exists - if not, reset complete status file
+        if os.path.isdir(self.complete_status_file_path):
+            self.composite_logger.log_error("Core state file path returned a directory. Attempting to reset.")
+            shutil.rmtree(self.complete_status_file_path)
+        if not os.path.exists(self.complete_status_file_path) and initial_load:
             self.composite_logger.log_warning("Status file not found at initial load. Resetting status file to defaults.")
             self.__reset_status_file()
             return
 
-        # Read the status file - raise exception on persistent failure
-        for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
-            try:
-                with self.env_layer.file_system.open(self.status_file_path, 'r') as file_handle:
-                    status_file_data_raw = json.load(file_handle)[0]    # structure is array of 1
-            except Exception as error:
-                if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT - 1:
-                    time.sleep(i + 1)
-                else:
-                    self.composite_logger.log_error("Unable to read status file (retries exhausted). Error: {0}.".format(repr(error)))
-                    raise
+        # Read the complete status file - raise exception on persistent failure
+        # Remove old complete status files and retain latest version
+        latest_complete_status_file_path = self.__get_latest_complete_status_file_path(self.execution_config.status_folder)
```

Comment:
Multiple pointers here:
1. The function name __get_latest_complete_status_file_path() does not fully explain it's actions. Name only suggests, this function is fetching the latest complete status file, whereas the function is also deleting other files.
2 This function determines latest by sorting all files based on highest sequence number. Why use this logic? 
Instead, since the naming pattern for status files is always <seq no>.complete.status, the latest .complete.status should always be the current <seq_no>.complete.status and there will only be 1 file with this name. We can simply remove all other files. I do not see the need to fetch latest based on a computation